disdrodb/data_transfer/download_data.py

```diff
@@ -0,0 +1,282 @@
+        Unzip the file if it is a zip file, by default False
+    """
+
+    local_path, url = url_local_path
+    if check_url(url):
+        # create station folder if not existing
+        if not os.path.exists(local_path):
+            os.makedirs(local_path)
+        # download the file
+        download_file_from_url(url, local_path)
+        print(f"Download {url} into {local_path}")
+        if url.endswith(".zip") and unzip:
+            zip_local_path = os.path.join(local_path, os.path.basename(url))
+            unzip_file(zip_local_path, local_path)
+            os.remove(zip_local_path)
+
+    else:
+        # raise an error if url not well formated
+        raise ValueError(f"URL {url} is not well formated. Please check the URL.")
+
+
+def download_all_files(
+    raw_folder: str = None,
+    data_source: str = None,
+    campaign_name: str = None,
+    station_name: str = None,
+    overwrite: bool = False,
+    unzip: bool = False,
+):
+    """Batch function to get all yaml files that coantain
+    the 'data_url' key and download the data locally.
+
+    Parameters
+    ----------
+    raw_folder : str, optional
+        Raw data folder path.
+        Must end with DISDRODB/Raw
+    data_source : str, optional
+        Data source folder name (eg : EPFL).
+        If not provided (None), all data sources will be downloaded.
+        The default is data_source=None
+    campaign_name : str, optional
+        Campaign name (eg :  EPFL_ROOF_2012).
+        If not provided (None), all campaigns will be downloaded.
+        The default is campaign_name=None
+    station_name : str, optional
+        Station name.
+        If not provided (None), all stations will be downloaded.
+        The default is station_name=None
+    overwrite : bool, optional
+        If True, overwrite the already existing raw data file.
+        The default is False.
+    unzip : bool, optional
+        If True, unzip the zip file.
+        The default is False.
+    """
+
+    list_urls_local_paths = get_list_urls_local_paths(raw_folder, data_source, campaign_name, station_name)
+
+    for url_local_path in list_urls_local_paths:
+        download_file_content(url_local_path, overwrite, unzip)
+
+
+@click.command()
+@click.option(
+    "--raw_folder",
+    required=True,
+    help="Raw data folder path (eg : /home/user/DISDRODB/Raw). Is compulsory.",
+)
+@click.option(
+    "--data_source",
+    help="Data source folder name (eg : EPFL). If not provided (None), all data sources will be downloaded.",
+)
+@click.option(
+    "--campaign_name",
+    help="Name of the campaign (eg :  EPFL_ROOF_2012). If not provided (None), all campaigns will be downloaded.",
+)
+@click.option(
+    "--station_name",
+    help="Station name. If not provided (None), all stations will be downloaded.",
+)
+@click.option("--overwrite", type=bool, help="Overwite existing file ?")
+@click.option("--unzip", type=bool, help="Unzip  existing file ?")
+def download_file(

```

Comment:
Make no sense to add this function here. This is a script that we want to be able to launch from the command line ! 
To do this: 
1. Add `scripts` directory into `data_transfer` 
2. Add a python file called `download_disdrodb_archive.py`
3. Put inside this function (renamed `download_disdrodb_archive`) (see scripts files into disdrodb/l0/scripts for example)
4. Add entry points for download_disdrodb_archive into the `pyproject.toml `under the `[project.scripts]`

Same process to be done for the tools to upload the data. Something like: 
`upload_disdrodb_archive`